### PR TITLE
[Impeller] Assign incremental clip depth to all entities.

### PR DESCRIPTION
**Pull request by bdero**
- Add depth settings to pipeline variant hash.
- Assign new clip depth to all entities.
- Punt the final depth to all the vertex shaders.
- Add validations to assert all clip depths are resolved.

### CODE REVIEW

**Comment by bdero** [comment]
This gets us as close to switching clips to the depth buffer as possible without actually doing it. I want to keep the diff to flip it on minimal so that we can easily revert it if there are problems.

**Comment by bdero** [comment]
Got some weird stuff going on with the goldens again. Looking into it.

**Comment by bdero** [comment]
nvm, accidentally changed the compare op, so this should be fixed.

File: impeller/entity/entity.cc [hunk 1]
```diff
@@ -86,14 +91,28 @@ const std::shared_ptr<Contents>& Entity::GetContents() const {
...
 
+void Entity::SetNewClipDepth(uint32_t clip_depth) {
+  new_clip_depth_ = clip_depth;
+}
+
+uint32_t Entity::GetNewClipDepth() const {
+  return new_clip_depth_;
+}
+
+static Scalar depth_epsilon = 1.0f / std::pow(2, 18);
```
Comment (jonahwilliams) [inline impeller/entity/entity.cc:110]: constexpr? or at least const

File: impeller/entity/entity_playground.cc [hunk 2]
```diff
@@ -30,6 +30,8 @@ bool EntityPlayground::OpenPlaygroundHere(EntityPass& entity_pass) {
     return false;
   }
 
+  entity_pass.PopAllClips(1000);
```
Comment (jonahwilliams) [inline impeller/entity/entity_playground.cc:33]: ?

File: impeller/renderer/backend/gles/texture_gles.cc [hunk 3]
```diff
@@ -154,12 +154,12 @@ struct TexImage2DData {
         // only use the stencil component.
         //
         // https://registry.khronos.org/OpenGL-Refpages/gl4/html/glTexImage2D.xhtml
+      case PixelFormat::kD24UnormS8Uint:
```
Comment (jonahwilliams) [inline impeller/renderer/backend/gles/texture_gles.cc:157]: hmm, should we have been hitting this path before?

**Review by jonahwilliams — APPROVED** [review]
LGTM

For my understanding, this adjusts the depth but we're still clipping via stencil ops, right?

File: impeller/entity/entity.cc (hunk 1 above)
Comment (bdero) [reply impeller/entity/entity.cc:110]: Done.

File: impeller/renderer/backend/gles/texture_gles.cc (hunk 3 above)
Comment (bdero) [reply impeller/renderer/backend/gles/texture_gles.cc:157]: I don't think we ever hit this path if offscreen MSAA is supported.

**Comment by bdero** [comment]
> For my understanding, this adjusts the depth but we're still clipping via stencil ops, right?

Yeah, it gets the correct depth values pumping into clip space but doesn't actually use them yet.

**Pull request merged**